### PR TITLE
Documentation refactoring

### PR DESCRIPTION
**Pull request by Bituvo**
**Ready for review (and possibly becoming a 5.9 milestone)!**

This PR adds these things to the documentation:
 - Modern primary and secondary headings
 - Capitalized headers
 - Slightly less whitespace before primary headers
 - Github flavored warnings and information boxes
 - Lua code blocks where appropriate
 - Tables where appropriate
 - Inline code blocks where appropriate
 - Removal of extra whitespace in tables (`{ min=3, max=5 }` -> `{min=3, max=5}`)
 - Typo fixes (especially `id` instead of `ID`)
 - Removed table of contents from `/README.md` as Github has a builtin one

To do (preferably in a different PR, just putting it out there):
 - Use a consistent maximum line width (maybe increasing with each level of bullet points)
 - Scour for typos and other mistakes
 - Redo `protocol.txt`

Some people still use the plaintext version of the Lua API, so I tried to maintain readability while still simplifying things.
The original PR for this was #13494, but it was messed up because I forgot to create a new branch.

### CODE REVIEW

**Comment by appgurueu** [comment]
Regarding the table of contents (ToC), the same as for the other PR applies:

> I feel like such a table of contents is (1) redundant and (2) probably will eventually be out-of-sync with the document. I think generating a ToC is the job of Markdown renderers.

And as ruben pointed out, "GitHub shows a table of contents in the top left."

File: doc/android.md [hunk 1]
```diff
@@ -10,32 +10,28 @@ due to limited capabilities of common devices. What can be done is described bel
...
 * Buttons:
-
-1. Left upper corner: Chat
-2. Right lower corner: Jump
-3. Right lower corner: Crouch
-4. Left lower corner (Joystick): Walk/step...
-5. Left lower corner: Display inventory
+  1. Upper left corner: Chat
+  2. Lower right corner: Jump/Crouch
+  4. Lower left corner (Joystick): Walk/step/Display inventory
```
Comment (appgurueu) [inline doc/android.md:21]: 3

Comment (Bituvo) [reply doc/android.md]: Fixed.

**Review by AFCMS — COMMENTED** [review]
This is something I really like.

I considered it when making #13439

File: doc/compiling/README.md [hunk 2]
```diff
@@ -1,42 +1,41 @@
...
 
-- [Compiling on GNU/Linux](linux.md)
-- [Compiling on Windows](windows.md)
-- [Compiling on MacOS](macos.md)
-
+* [Compiling on GNU/Linux](linux.md)
+* [Compiling on Windows](windows.md)
+* [Compiling on MacOS](macos.md)
 
 ## CMake options
```
Comment (Bituvo) [inline doc/compiling/README.md:7]: Capitalize

File: doc/compiling/windows.md [hunk 3]
```diff
@@ -2,32 +2,29 @@
...
-- [CMake](https://cmake.org/download/)
-- [vcpkg](https://github.com/Microsoft/vcpkg)
-- [Git](https://git-scm.com/downloads)
-
+* [Visual Studio 2015 or newer](https://visualstudio.microsoft.com)
+* [CMake](https://cmake.org/download/)
+* [vcpkg](https://github.com/Microsoft/vcpkg)
+* [Git](https://git-scm.com/downloads)
 
 ## Compiling and installing the dependencies
```
Comment (Bituvo) [inline doc/compiling/windows.md:10]: Capitalize

File: doc/compiling/windows.md [hunk 4]
```diff
@@ -2,32 +2,29 @@
...
+* `gmp` and `jsoncpp` are optional, otherwise the bundled versions will be compiled
 
 There are other optional libraries, but they are not tested if they can build and link correctly.
 
 Use `--triplet` to specify the target triplet, e.g. `x64-windows` or `x86-windows`.
 
-
 ## Compile Minetest
 
 ### a) Using the vcpkg toolchain and CMake GUI
```
Comment (Bituvo) [inline doc/compiling/windows.md:30]: Capitalize

File: doc/menu_lua_api.md [hunk 5]
```diff
@@ -78,26 +64,41 @@ Filesystem
...
+  * `param`=true: returns path to a temporary file
+    otherwise: returns path to the temporary folder
+
+## Network
+
+* `core.get_min_supp_proto()`
+  * returns the minimum supported network protocol version
+* `core.get_max_supp_proto()`
+  * returns the maximum supported network protocol version
+## Media
```
Comment (Bituvo) [inline doc/menu_lua_api.md:78]: Spacing

File: doc/menu_lua_api.md [hunk 6]
```diff
@@ -78,26 +64,41 @@ Filesystem
...
 * `core.get_video_drivers()`
   * get list of video drivers supported by engine (not all modes are guaranteed to work)
   * returns list of available video drivers' settings name and 'friendly' display name
-    e.g. `{ {name="opengl", friendly_name="OpenGL"}, {name="software", friendly_name="Software Renderer"} }`
+    e.g. `{{name="opengl", friendly_name="OpenGL"}, {name="software", friendly_name="Software Renderer"}}`
   * first element of returned list is guaranteed to be the NULL driver
+
+
+
+# Misc.
```
Comment (Bituvo) [inline doc/menu_lua_api.md:91]: Spacing

File: doc/texture_packs.md [hunk 7]
```diff
@@ -1,11 +1,15 @@
...
+# Minetest Texture Pack Reference
 
 Texture packs allow you to replace textures provided by a mod with your own
 textures.
 
-Texture pack directory structure
---------------------------------
+
+
+# Texture pack directory structure
```
Comment (Bituvo) [inline doc/texture_packs.md:8]: Capitalize

File: doc/texture_packs.md [hunk 8]
```diff
@@ -48,19 +48,21 @@ The custom textures do not necessarily require the same size as their
...
+
+
+# Special textures
 
 These texture names are hardcoded into the engine but can also be overwritten
 by texture packs. All existing fallback textures can be found in the directory
 `textures/base/pack`.
 
-### Gameplay textures
+## Gameplay textures
```
Comment (Bituvo) [inline doc/texture_packs.md:65]: Capitalization

File: doc/texture_packs.md [hunk 9]
```diff
@@ -131,7 +133,7 @@ are placeholders intended to be overwritten by the game.
 * `server_flags_favorite.png`: icon for your favorite servers
 * `server_flags_pvp.png`: icon for enabled PvP on servers
 
-### Android textures
+## Android textures
```
Comment (Bituvo) [inline doc/texture_packs.md:136]: Spacing

**Review by Bituvo — COMMENTED** [review]
Self-review

File: doc/lua_api.md [hunk 10]
```diff
@@ -7483,17 +7481,15 @@ child will follow movement and rotation of that bone.
         * Fourth column: subject looking to the right
         * Fifth column: subject viewed from above
         * Sixth column: subject viewed from below
-* `get_entity_name()`
-  > **Warning**: Deprecated, will be removed in a future version, use the field `self.name` instead
+* `get_entity_name()`: (**Deprecated**: Will be removed in a future version, use the field `self.name` instead)
 * `get_luaentity()`
 
 #### Player Only (No-Op for Other Objects)
```
Comment (appgurueu) [inline doc/lua_api.md:7487]: The syntax highlighting here is off. Could this be due to an unclosed Lua code block?

Comment (Bituvo) [reply doc/lua_api.md:7487]: Weird, this doesn't show in VS Code. I don't see any open code blocks, either.
image

Comment (appgurueu) [reply doc/lua_api.md:7487]: Could also be a GitHub bug then.

**Comment by rollerozxa** [comment]
There's quite a lot of noise in the PR caused by ".  " being replaced with ". " in things such as code comments. (as well as translation files which I think shouldn't be touched?)

File: README.md [hunk 11]
```diff
@@ -10,30 +9,22 @@ Minetest is a free open-source voxel game engine with easy modding and game crea
...
-- Forum: https://forum.minetest.net/
-- GitHub: https://github.com/minetest/minetest/
-- [doc/](doc/) directory of source distribution
-
-Default controls
-----------------
+## In Case You Downloaded the Source Code
+
+If you downloaded the Minetest Engine source code in which this file is
+contained, you probably want to download the [Minetest Game](https://github.com/minetest/minetest_game/)
```
Comment (rollerozxa) [inline README.md:15]: merge blemish

File: doc/lua_api.md [hunk 12]
```diff
@@ -2091,7 +2084,7 @@ An example recipe: Craft a raw meat soup from any meat, any water and any bowl:
         {"group:meat"},
         {"group:water"},
         {"group:bowl"},
-    },
+,    }
```
Comment (rollerozxa) [inline doc/lua_api.md:2087]: typo?

File: doc/lua_api.md [hunk 13]
```diff
@@ -2368,28 +2348,27 @@ Table of resulting tool uses:
...
 Damage calculation:
 
-    damage = 0
-    foreach group in cap.damage_groups:
-        damage += cap.damage_groups[group]
-            * limit(actual_interval / cap.full_punch_interval, 0.0, 1.0)
-            * (object.armor_groups[group] / 100.0)
-            -- Where object.armor_groups[group] is 0 for inexistent values
-    return damage
+```lua
```
Comment (rollerozxa) [inline doc/lua_api.md:2363]: this looks to be some sort of python-lua hybrid pseudocode. could change it to be more in line with lua to make the syntax highlighting better

File: README.md (hunk 11 above)
Comment (Bituvo) [reply README.md:15]: What?

File: doc/lua_api.md (hunk 13 above)
Comment (Bituvo) [reply doc/lua_api.md:2363]: I'm not really sure how the damage calculation works.

File: README.md (hunk 11 above)
Comment (rollerozxa) [reply README.md:15]: sorry, I meant that this got mistakenly left in when merge conflicts were resolved, it doesn't exist in the master branch anymore

File: .github/CONTRIBUTING.md [hunk 14]
```diff
@@ -2,72 +2,73 @@
...
 
-   Any Pull Request that isn't a bug fix and isn't covered by
+   Any pull request that isn't a bug fix and isn't covered by
    [the roadmap](../doc/direction.md) will be closed within a month unless it
    receives a concept approval from a Core Developer. For this reason, it is
    recommended that you open an issue for any such pull requests before doing
    the work, to avoid disappointment.
 
    You may also benefit from discussing on our IRC development channel
-   [#minetest-dev](http://www.minetest.net/irc/). Note that a proper IRC client
```
Comment (rollerozxa) [inline .github/CONTRIBUTING.md:28]: This page doesn't exist anymore and redirects to `/get-involved/`, https://wiki.minetest.net/IRC would probably be a better page to link to.

File: .github/CONTRIBUTING.md [hunk 15]
```diff
@@ -122,20 +123,20 @@ translated by editing a `.tr` text file. See
 ## Donations
 
 If you'd like to monetarily support Minetest development, you can find donation
-methods on [our website](http://www.minetest.net/development/#donate).
+methods on [our website](https://www.minetest.net/development/#donate).
```
Comment (rollerozxa) [inline .github/CONTRIBUTING.md:126]: `https://www.minetest.net/get-involved/#donate`

File: README.md (hunk 11 above)
Comment (Bituvo) [reply README.md:15]: Ok, will remove

File: .github/CONTRIBUTING.md (hunk 14 above)
Comment (Bituvo) [reply .github/CONTRIBUTING.md:28]: Done, and reworded a bit.

File: doc/lua_api.md (hunk 13 above)
Comment (Bituvo) [reply doc/lua_api.md:2363]: Fixed!

**Comment by Zughy** [comment]
@Bituvo rebase needed

**Comment by Bituvo** [comment]
> Actually, according to https://irc.minetest.net/minetest-dev/2024-01-15#i_6146082, I don't this PR will make it. If split into smaller PRs and if covering only relevant aspects (see Rollerozxa comment), I guess there won't be big issues though

I completely understand. A new PR has been opened, much cleaner and with less noise: #14287

**Pull request closed without merging**